uaclient/entitlements/__init__.py

```diff
@@ -23,12 +24,14 @@
 ]  # type: List[Type[UAEntitlement]]
 
 
-ENTITLEMENT_CLASS_BY_NAME = dict(
-    (cast(str, cls.name), cls) for cls in ENTITLEMENT_CLASSES
-)  # type: Dict[str, Type[UAEntitlement]]
+def entitlement_factory(name: str) -> UAEntitlement:
+    for entitlement in ENTITLEMENT_CLASSES:
+        if name in entitlement().valid_names:
+            return entitlement
+    return None
```

Comment:
I think that your creation of a factory function here is good and explicit even though it involved having to add a "valid_names" method on classes and call that method inside `valid_services` function.

An alternative to dropping ENTITLEMENT_CLASS_BY_NAME would be defining aliases keys inside  ENTITLEMENT_CLASS_BY_NAME as such:
```python
{
 'usg': CISEntitlement,
 'cis': CISEntitlement,
} 
```
Having aliases represented in  ENTITLEMENT_CLASS_BY_NAME you'd be able to avoid the `Entitlement.valid_names` class method I think, but you'd add complexity to all call-sites (like `valid_services`) to walk the whole ENTITLEMENT_CLASS_BY_NAME dict and compare values to find all valid_names for a given entitlement. The alternative I suggest seems a bit more implicit a construct than what you have, so I agree with your approach here.